pkg/daemon/certs/certs.go

```diff
@@ -196,3 +303,47 @@ func addCertToStore(store windows.Handle, cert *x509.Certificate) error {
 	klog.Infof("Certificate %s imported successfully.", cert.Subject.String())
 	return nil
 }
+
+// removeCertFromStore removes the given certificate from its associated certificate store
+func removeCertFromStore(store windows.Handle, cert *x509.Certificate) error {
+	// Windows CertContext holds the store which imported the cert, which is needed when deleting from the system store
+	certContext, err := retrieveCertContext(store, cert)
+	if err != nil {
+		return fmt.Errorf("Failed to create contet from x509 cert %s: %w\n", cert.Subject.String(), err)
+	}
+	if err = windows.CertDeleteCertificateFromStore(certContext); err != nil {
+		return fmt.Errorf("Failed to remove certificate %s: %z\n", cert.Subject.String(), err)
+	}
+	klog.Infof("Certificate %s removed successfully.", cert.Subject.String())
+	return nil
+}
+
+// retrieveCertContext finds the given certificate in the given store and returns its full context per Windows OS
+func retrieveCertContext(store windows.Handle, targetCert *x509.Certificate) (*windows.CertContext, error) {
+	var certContext *windows.CertContext
+	var err error
+	for {
+		certContext, err = windows.CertEnumCertificatesInStore(store, certContext)
+		if err != nil {
+			if errors.Is(err, windows.Errno(windows.CRYPT_E_NOT_FOUND)) {
+				// Error code implies we have read all certs:
+				// https://learn.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certenumcertificatesinstore
+				break
+			}
+			return nil, fmt.Errorf("Error reading existing system certificate: %w", err)
+		}
+
+		// Convert the certificate bytes to Golang x509.Certificate
+		certBytes := unsafe.Slice(certContext.EncodedCert, certContext.Length)
+		current, err := x509.ParseCertificate(certBytes)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to parse certificate from bytes: %w", err)
+		}
+
+		if current.Equal(targetCert) {
+			klog.Infof("found %s existing system certificate", certContext.CertInfo.Subject)
+			return certContext, nil
+		}
+	}
+	return nil, fmt.Errorf("unable to find target cert %s in root system store", targetCert.Subject)
```

Comment:
From the error message I take it that the cert we want to delete is not in the system store. Doesn't that imply it is not present? Shouldn't we just log and move on.

Reply:
Fair point, I'll log instead of returning an error